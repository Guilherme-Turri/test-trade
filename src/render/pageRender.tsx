import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};