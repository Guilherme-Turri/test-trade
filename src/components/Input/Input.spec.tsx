/* eslint-disable testing-library/no-unnecessary-act */
import { componentRender } from "../../render/componentRender";
import { screen, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import { Input } from ".";

describe('<Input />', () => {
  const mockValues = {
    type: 'text',
    name: 'nome',
    onChange: jest.fn(),
    error: '',
    value: ''
  }
  it('should render on screen properly', () => {
    const { container } = componentRender(<Input {...mockValues} />)
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })

  it('should render on screen whith error message', () => {
    mockValues.error = 'errorMsg'
    componentRender(<Input {...mockValues} />)
    const input = screen.getByTestId('Input');
    act(() => {
      userEvent.type(input, 'u');
      input.blur();
    });
    const errorMessage = screen.getByText('errorMsg');
    expect(errorMessage).toBeInTheDocument();
  })
})