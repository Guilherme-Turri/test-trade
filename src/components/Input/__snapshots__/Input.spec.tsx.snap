// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render on screen properly 1`] = `
.c0 {
  outline: none;
  all: unset;
  border: 1px solid #00c555;
  border-radius: 0.313rem;
  width: 100%;
  height: 1.563rem;
  padding: 0.938rem;
  box-sizing: border-box;
  color: #2d5839;
}

.c0::-webkit-input-placeholder {
  color: '#95A5A0'!important;
}

.c0::-moz-placeholder {
  color: '#95A5A0'!important;
}

.c0:-ms-input-placeholder {
  color: '#95A5A0'!important;
}

.c0::placeholder {
  color: '#95A5A0'!important;
}

<div>
  <input
    autocomplete="new-password"
    class="c0"
    data-testid="Input"
    placeholder="nome"
    type="text"
    value=""
  />
</div>
`;
